tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_OverlapHelpers.cpp

```diff
@@ -56,6 +179,167 @@ SPECTRE_TEST_CASE("Unit.ParallelSchwarz.OverlapHelpers",
     CHECK(overlap_width(3, coords) == approx(1.8));
     CHECK(overlap_width(4, coords) == approx(2.));
   }
+  {
+    INFO("Overlap iterator");
+    // Note: The iterator advances slice-by-slice in the overlap dimension
+    // Overlap region: [X X O] -xi->
+    test_overlap_iterator(Index<1>{3}, 2, Direction<1>::lower_xi(), {{0, 1}},
+                          {{0, 1}});
+    // Overlap region: [O X X] -xi->
+    test_overlap_iterator(Index<1>{3}, 2, Direction<1>::upper_xi(), {{1, 2}},
+                          {{0, 1}});
+    // Overlap region:
+    // + - - - + -xi->
+    // | X X O |
+    // | X X O |
+    // + - - - +
+    // v eta
+    test_overlap_iterator(Index<2>{{{3, 2}}}, 2, Direction<2>::lower_xi(),
+                          {{0, 3, 1, 4}}, {{0, 2, 1, 3}});
+    // Overlap region:
+    // + - - - + -xi->
+    // | O X X |
+    // | O X X |
+    // + - - - +
+    // v eta
+    test_overlap_iterator(Index<2>{{{3, 2}}}, 2, Direction<2>::upper_xi(),
+                          {{1, 4, 2, 5}}, {{0, 2, 1, 3}});
+    // Overlap region:
+    // + - - - - + -xi->
+    // | X X X X |
+    // | X X X X |
+    // | O O O O |
+    // + - - - - +
+    // v eta
+    test_overlap_iterator(Index<2>{{{4, 3}}}, 2, Direction<2>::lower_eta(),
+                          {{0, 1, 2, 3, 4, 5, 6, 7}},
+                          {{0, 1, 2, 3, 4, 5, 6, 7}});
+    // Overlap region:
+    // + - - + -xi->
+    // | O O |
+    // | X X |
+    // | X X |
+    // + - - +
+    // v eta
+    test_overlap_iterator(Index<2>{{{2, 3}}}, 2, Direction<2>::upper_eta(),
+                          {{2, 3, 4, 5}}, {{0, 1, 2, 3}});
+    test_overlap_iterator(Index<3>{{{3, 3, 2}}}, 2, Direction<3>::lower_xi(),
+                          {{0, 3, 6, 9, 12, 15, 1, 4, 7, 10, 13, 16}},
+                          {{0, 2, 4, 6, 8, 10, 1, 3, 5, 7, 9, 11}});
+    test_overlap_iterator(Index<3>{{{3, 3, 2}}}, 2, Direction<3>::upper_xi(),
+                          {{1, 4, 7, 10, 13, 16, 2, 5, 8, 11, 14, 17}},
+                          {{0, 2, 4, 6, 8, 10, 1, 3, 5, 7, 9, 11}});
+    test_overlap_iterator(Index<3>{{{3, 3, 2}}}, 2, Direction<3>::lower_eta(),
+                          {{0, 1, 2, 9, 10, 11, 3, 4, 5, 12, 13, 14}},
+                          {{0, 1, 2, 6, 7, 8, 3, 4, 5, 9, 10, 11}});
```

Comment:
(Picking an arbitrary example.) I don't know if it actually matters in your intended uses, but you may get better performance if the iterator produces the indices in order so your array accesses jump around less.

Reply:
The performance might matter. I made the order of indices sequential in a fixup. Thanks for suggesting.